src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java

```diff
@@ -132,6 +132,28 @@ public void testLiteralFor() throws Exception {
     }
 
     @Test
+    public void testLiteralDo() throws Exception {
+        final DefaultConfiguration configurationTestLiteralDo =
+                createCheckConfig(WhitespaceAfterCheck.class);
+        configurationTestLiteralDo.addAttribute("tokens", "LITERAL_DO");
+        final String[] expected = {
+            "63:11: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "do"),
+        };
+        verify(configurationTestLiteralDo, getPath("InputWhitespaceAfter.java"), expected);
+    }
+
+    @Test
+    public void testDoWhile() throws Exception{
+        final DefaultConfiguration configurationTestDoWhile =
+                createCheckConfig(WhitespaceAfterCheck.class);
+        configurationTestDoWhile.addAttribute("tokens", "DO_WHILE");
+        final String[] expected = {
+            "18:16 " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "do{}while"),
```

Comment:
@MaksimP , what is "do{}while" ? you expect space after "while" (the fact that it is do-while is known only to AST tree), but violation is on "while" token.